src/EventListener/AuditLogListener/AuditLogClient/SymfonyHttpAuditLogClient.php

```diff
@@ -0,0 +1,44 @@
+<?php
+declare(strict_types=1);
+
+namespace Landingi\EventStoreBundle\EventListener\AuditLogListener\AuditLogClient;
+
+use Landingi\EventStoreBundle\EventListener\AuditLogListener\AuditLogClient;
+use Landingi\EventStoreBundle\EventListener\AuditLogListener\AuditLogEvent;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+final class SymfonyHttpAuditLogClient implements AuditLogClient
+{
+    private string $auditLogApiBaseUrl;
+    private HttpClientInterface $auditLogApiHttpClient;
+
+    public function __construct(string $auditLogApiBaseUrl, HttpClientInterface $auditLogApiHttpClient)
+    {
+        $this->auditLogApiBaseUrl = $auditLogApiBaseUrl;
+        $this->auditLogApiHttpClient = $auditLogApiHttpClient;
+    }
+
+    public function store(AuditLogEvent $event): void
+    {
+        $this->auditLogApiHttpClient->request(
+            'POST',
+            "{$this->auditLogApiBaseUrl}/v1/events",
+            [
+                'json' => [
```

Comment:
Możesz tutaj użyć serializacji i niepotrzeba Ci getterów i 10 linijek papa

Reply:
myślałem o tym, ale stwierdziłem, że jak to wyniose do jsonSerialize w obiekcie AuditLogEvent  to bede miał tam to samo czyli dane skrojone pod API auditloga co mnie sie wydaje mieszaniem odpowiedzialnosci bo jak zmienie API to moim zdaniem kod klienta sie powinien dostosowac a nie tego obiektu

Reply:
Masz rację, ale w tym soft to AuditLogEvent jest w zasadzie DTO XD... chyba żeby jakiegos mappera zrobić albo printera? W zasadzie to byłby legitny case ^^

Reply:
@Ferror @SBielawski jaki kurła mapper mam tutaj dodawać i po co :shrug:

Reply:
kolejny DTO do DTO to będzie, przecież te obiekty nie mają zachowań, to są tak na prawdę tablice niemutowalne XD

Reply:
mogę zrobić obiekt HttpAuditLogEvent z jsonSerialize pod ten case do API clienta po http